control/phaseplot.py

```diff
@@ -1,61 +1,912 @@
+            kwargs, plot_equilpoints, gridspec=gridspec, ax=ax)
+        out[2] = equilpoints(
+            sys, pointdata, check_kwargs=False, **kwargs_local)
+
+        # Get rid of keyword arguments handled by equilpoints
+        for kw in ['params']:
+            initial_kwargs.pop(kw, None)
+
+    # Make sure all keyword arguments were used
+    if initial_kwargs:
+        raise TypeError("unrecognized keywords: ", str(initial_kwargs))
+
+    if fig is not None:
+        fig.suptitle(f"Phase portrait for {sys.name}")
+        ax.set_xlabel(sys.state_labels[0])
+        ax.set_ylabel(sys.state_labels[1])
+
+    return out
+
+
+def vectorfield(
+        sys, pointdata, gridspec=None, ax=None, check_kwargs=True, **kwargs):
+    """Plot a vector field in the phase plane.
+
+    This function plots a vector field for a two-dimensional state
+    space system.
+
+    Parameters
+    ----------
+    sys : NonlinearIOSystem or callable(x, t, ...)
+        Function used to generate phase plane data.
+    pointdata : list or 2D array
+        List of the form [xmin, xmax, ymin, ymax] describing the
+        boundaries of the phase plot or an array of shape (N, 2)
+        giving points of at which to plot the vector field.
+    gridtype : str, optional
+        The type of grid to use for generating initial conditions:
+        'meshgrid' (default) generates a mesh of initial conditions within
+        the specified boundaries, 'boxgrid' generates initial conditions
+        along the edges of the boundary, 'circlegrid' generates a circle of
+        initial conditions around each point in point data.
+    gridspec : list, optional
+        If the gridtype is 'meshgrid' and 'boxgrid', `gridspec` gives the
+        size of the grid in the x and y axes on which to generate points.
+        If gridtype is 'circlegrid', then `gridspec` is a 2-tuple
+        specifying the radius and number of points around each point in the
+        `pointdata` array.
+    color : str
+        Plot the vector field in the given color.
+    ax : Axes
+        Use the given axes for the plot, otherwise use the current axes.
+
+    Returns
+    -------
+    out : Quiver
+
+    """
+    # Get system parameters
+    params = kwargs.pop('params', None)
+
+    # Determine the points on which to generate the vector field
+    points, _ = _make_points(pointdata, gridspec, 'meshgrid')
+
+    # Create axis if needed
+    if ax is None:
+        ax = plt.gca()
+
+    # Set the plotting limits
+    xlim, ylim, maxlim = _set_axis_limits(ax, pointdata)
+
+    # Figure out the color to use
+    color = _get_color(kwargs, ax)
+
+    # Make sure all keyword arguments were processed
+    if check_kwargs and kwargs:
+        raise TypeError("unrecognized keywords: ", str(kwargs))
+
+    # Generate phase plane (quiver) data
+    vfdata = np.zeros((points.shape[0], 4))
+    for i, x in enumerate(points):
+        vfdata[i, :2] = x
+        vfdata[i, 2:] = sys.dynamics(0, x, 0, params=params)
```

Comment:
I think this wouldn't work if `sys` is just a function - is the callable intended to be a function if it is not a system?